### ISSUE TITLE
Formatting Commands

### ISSUE THREAD

**Issue by Mark-Phillipson** [cursorless-dev/cursorless#2225]
Given that people use Cursorless for dictating normal text, would it be possible, if this is not already in the pipeline, to enhance the formatting commands in Cursorless.

Currently for example we have to say:

format title at [target]

Proposed New Commands:
```
title [target]
camel [target]
smash [target]
all down [target]
all up [target]
sentence [target]
hammer [target]
dotted [target]

```
This would make the commands more intuitive and easier to remember. Being able to combine some of the commands also could make sense. For example:
`dotted title [target] `

**Comment by pokey** [comment]
I'm hesitant due to the conflicts with the built-in formatters. For a start would be better to do `"camel format <target>"`, but even that technically has conflicts

@AndreasArvidsson what are you currently using?

We should also expose format action in our Talon-side API so users can use whatever they want (https://github.com/cursorless-dev/cursorless/issues/492)

**Comment by AndreasArvidsson** [comment]
`camel form <target>`
https://github.com/AndreasArvidsson/cursorless-talon/blob/8d1108549f77e85dc746afbbd5ab4840542441dd/src/cursorless.talon#L18-L19

**Comment by pokey** [comment]
> `camel form <target>` https://github.com/AndreasArvidsson/cursorless-talon/blob/8d1108549f77e85dc746afbbd5ab4840542441dd/src/cursorless.talon#L18-L19

Interesting. Does it ever misfire or pretty reliable? I also wonder about DFA

**Comment by AndreasArvidsson** [comment]
> > `camel form <target>` https://github.com/AndreasArvidsson/cursorless-talon/blob/8d1108549f77e85dc746afbbd5ab4840542441dd/src/cursorless.talon#L18-L19
> 
> Interesting. Does it ever misfire or pretty reliable? I also wonder about DFA

Not one hundred percent reliable, but that is true about a bunch of other commands I used daily. Personally I prefer it this way otherwise of course I would just revert to the default. DFA is fine

**Comment by pokey** [comment]
Would you recommend it for default, or should we just add format command to api and add a recipe to wiki?

**Comment by AndreasArvidsson** [comment]
Yes I think making this default would be reasonable. You should probably try it for yourself first?

**Comment by fidgetingbits** [comment]
fwiw I've used the same as Andreas for months and rarely (if ever) have problems.

**Comment by AndreasArvidsson** [comment]
fyi this is what I'm using
```talon
<user.formatters> {user.cursorless_reformat_action} <user.cursorless_target>:
    user.cursorless_reformat(cursorless_target, formatters)
```

**Comment by pokey** [comment]
@AndreasArvidsson is this using public api's? If not, should we make all these things public?

**Comment by AndreasArvidsson** [comment]
Yes it is
